tools/src/main/scala/scala/scalanative/linker/Assembly.scala

```diff
@@ -16,14 +14,16 @@ object Assembly {
   final case class Dir private[Assembly](val base: File) extends Assembly {
     //println(s"discovered dir assembly $base")
 
+    def matchFiles(f: File): Array[File] = {
+      if (f.isDirectory) f.listFiles.map(matchFiles).flatten.toArray
+      else if (f.getName.endsWith(".nir")) Array(f)
+      else Array()
+    }
```

Comment:
To be honest I didn't think too much at "optimization" at this stage, you guys are crazy :-)

If you really care probably best option is to use java8 File interface with something like:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#walk-java.nio.file.Path-java.nio.file.FileVisitOption...-

Too me removing apache commons serves to speed up docker image creation :-) so pick your choice!